Module name cannot be determined if module declaration is annotated

module-info.java:
```
import org.jspecify.annotations.NullMarked;

@NullMarked
module org.beryx.modular.annotatedmodule {
    requires org.jspecify;
    opens org.beryx.modular.annotatedmodule;
}
```

Error with shortened stacktrace:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':prepareModulesDir'.
> Error while evaluating property 'moduleName' of task ':prepareModulesDir'.
   > Failed to query the value of extension 'jlink' property 'moduleName'.
      > Cannot retrieve module name from /private/var/folders/6w/05drv2hn6c731ccrddrcxb7r0000gn/T/spock_should_create_runti_0_testProjectDir5359671681591213750/src/main/java/module-info.java with content: import org.jspecify.annotations.NullMarked;
        
        @NullMarked
        module org.beryx.modular.annotatedmodule {
            requires org.jspecify;
            opens org.beryx.modular.annotatedmodule;
        }


* Try:
> Run with --debug option to get more log output.
> Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':prepareModulesDir'.
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:38)
	at ...
Caused by: org.gradle.api.internal.tasks.properties.PropertyEvaluationException: Error while evaluating property 'moduleName' of task ':prepareModulesDir'.
	at org.gradle.api.internal.tasks.properties.InputParameterUtils.prepareInputParameterValue(InputParameterUtils.java:33)
	at ...
Caused by: org.gradle.api.internal.provider.AbstractProperty$PropertyQueryException: Failed to query the value of extension 'jlink' property 'moduleName'.
	at org.gradle.api.internal.provider.AbstractProperty.doCalculateValue(AbstractProperty.java:136)
	at ...
Caused by: org.gradle.api.GradleException: Cannot retrieve module name from /private/var/folders/6w/05drv2hn6c731ccrddrcxb7r0000gn/T/spock_should_create_runti_0_testProjectDir5359671681591213750/src/main/java/module-info.java with content: import org.jspecify.annotations.NullMarked;

@NullMarked
module org.beryx.modular.annotatedmodule {
    requires org.jspecify;
    opens org.beryx.modular.annotatedmodule;
}

	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at org.gradle.internal.classpath.intercept.CallInterceptorsSet$DecoratingCallSite.access$401(CallInterceptorsSet.java:129)

```

--- a/src/main/groovy/org/beryx/jlink/util/Util.groovy
+++ b/src/main/groovy/org/beryx/jlink/util/Util.groovy
@@ -67,7 +67,11 @@ class Util {
 
     private static final String IMPORT_DECLARATION = 'import' + IGNORE + '(static' + IGNORE + ')?' + QUALIFIED_NAME + '(\\.\\*' + ')?' + IGNORE + ';'
     private static final String IMPORT_DECLARATIONS = '(' + IMPORT_DECLARATION + IGNORE + ')*'
-    private static final String MODULE_DECLARATION = '(?s)' + IGNORE + IMPORT_DECLARATIONS + '(open' + IGNORE + ')?' + 'module' + IGNORE + '(?<MODULE>' + QUALIFIED_NAME + ').*?'
+
+    private static final String ANNOTATION = '(@' + QUALIFIED_NAME + '((' + WS + ')*(\\([^)]*\\)))?)'
+    private static final String MODULE_ANNOTATIONS = '(' + ANNOTATION + IGNORE + ')*'
+
+    private static final String MODULE_DECLARATION = '(?s)' + IGNORE + IMPORT_DECLARATIONS + MODULE_ANNOTATIONS + '(open' + IGNORE + ')?' + 'module' + IGNORE + '(?<MODULE>' + QUALIFIED_NAME + ').*?'
 
     private static final Pattern PATTERN = Pattern.compile(MODULE_DECLARATION)
 

--- a/src/main/groovy/org/beryx/jlink/util/Util.groovy
+++ b/src/main/groovy/org/beryx/jlink/util/Util.groovy
@@ -68,7 +68,7 @@ class Util {
     private static final String IMPORT_DECLARATION = 'import' + IGNORE + '(static' + IGNORE + ')?' + QUALIFIED_NAME + '(\\.\\*' + ')?' + IGNORE + ';'
     private static final String IMPORT_DECLARATIONS = '(' + IMPORT_DECLARATION + IGNORE + ')*'
 
-    private static final String ANNOTATION = '(@' + QUALIFIED_NAME + '((' + WS + ')*(\\([^)]*\\)))?)'
+    private static final String ANNOTATION = '(@' + IGNORE + QUALIFIED_NAME + '((' + IGNORE + ')(\\([^)]*\\)))?)'
     private static final String MODULE_ANNOTATIONS = '(' + ANNOTATION + IGNORE + ')*'
 
     private static final String MODULE_DECLARATION = '(?s)' + IGNORE + IMPORT_DECLARATIONS + MODULE_ANNOTATIONS + '(open' + IGNORE + ')?' + 'module' + IGNORE + '(?<MODULE>' + QUALIFIED_NAME + ').*?'

--- a/src/test/groovy/org/beryx/jlink/JlinkPluginSpec.groovy
+++ b/src/test/groovy/org/beryx/jlink/JlinkPluginSpec.groovy
@@ -205,6 +205,21 @@ class JlinkPluginSpec extends Specification {
         checkOutput(result, 'helloBom', '{"from":"Alice","to":"Bob","greeting":"Hello"}')
     }
 
+    def "should create runtime image of project with annotations on module declaration"() {
+        when:
+        File buildFile = setUpBuild('hello-annotated-module')
+        BuildResult result = GradleRunner.create()
+                .withDebug(true)
+                .withGradleVersion('7.6')
+                .withProjectDir(testProjectDir.toFile())
+                .withPluginClasspath()
+                .withArguments(JlinkPlugin.TASK_NAME_JLINK, "-is")
+                .build();
+
+        then:
+        checkOutput(result, 'helloAnnotatedModule', 'Hello annotated module!')
+    }
+
     def "should create image of project with multiple launchers"() {
         when:
 

--- a/src/test/groovy/org/beryx/jlink/UtilSpec.groovy
+++ b/src/test/groovy/org/beryx/jlink/UtilSpec.groovy
@@ -51,6 +51,8 @@ class UtilSpec extends Specification {
         '/*my module*/\nmodule /*---*/ a.b.c // declaration\n{\n  exports a.b.c;\n}' | 'a.b.c'
         'import x.y.Z;//comment\nimport x.y.W;\nmodule /*---*/ a.b.c'                | 'a.b.c'
         'import x.y.z.*;\nimport x.y/*WW*/./*ZZ*/w.*;\nmodule //x.y.z\n/*-->*/a.b.c' | 'a.b.c'
+        'import x.y.z.*;\n@Annotation\nmodule //x.y.z\n/*-->*/a.b.c'                 | 'a.b.c'
+        'import x.y.z.*;\n@Annotation("text")\nmodule //x.y.z\n/*-->*/a.b.c'         | 'a.b.c'
     }
 
     @Unroll

--- a/src/test/groovy/org/beryx/jlink/UtilSpec.groovy
+++ b/src/test/groovy/org/beryx/jlink/UtilSpec.groovy
@@ -44,15 +44,17 @@ class UtilSpec extends Specification {
         Util.getModuleNameFrom(text) == moduleName
 
         where:
-        text                                                                         | moduleName
-        'module a.b.c\n{'                                                            | 'a.b.c'
-        'open module a.b.c{}'                                                        | 'a.b.c'
-        '  \t  open\t  \tmodule \ta.b.c\t { '                                        | 'a.b.c'
-        '/*my module*/\nmodule /*---*/ a.b.c // declaration\n{\n  exports a.b.c;\n}' | 'a.b.c'
-        'import x.y.Z;//comment\nimport x.y.W;\nmodule /*---*/ a.b.c'                | 'a.b.c'
-        'import x.y.z.*;\nimport x.y/*WW*/./*ZZ*/w.*;\nmodule //x.y.z\n/*-->*/a.b.c' | 'a.b.c'
-        'import x.y.z.*;\n@Annotation\nmodule //x.y.z\n/*-->*/a.b.c'                 | 'a.b.c'
-        'import x.y.z.*;\n@Annotation("text")\nmodule //x.y.z\n/*-->*/a.b.c'         | 'a.b.c'
+        text                                                                               | moduleName
+        'module a.b.c\n{'                                                                  | 'a.b.c'
+        'open module a.b.c{}'                                                              | 'a.b.c'
+        '  \t  open\t  \tmodule \ta.b.c\t { '                                              | 'a.b.c'
+        '/*my module*/\nmodule /*---*/ a.b.c // declaration\n{\n  exports a.b.c;\n}'       | 'a.b.c'
+        'import x.y.Z;//comment\nimport x.y.W;\nmodule /*---*/ a.b.c'                      | 'a.b.c'
+        'import x.y.z.*;\nimport x.y/*WW*/./*ZZ*/w.*;\nmodule //x.y.z\n/*-->*/a.b.c'       | 'a.b.c'
+        'import x.y.z.*;\n@Annotation\nmodule //x.y.z\n/*-->*/a.b.c'                       | 'a.b.c'
+        'import x.y.z.*;\n@Annotation("text")\nmodule //x.y.z\n/*-->*/a.b.c'               | 'a.b.c'
+        'import x.y.z.*;\n@ Annotation("text")\nmodule //x.y.z\n/*-->*/a.b.c'              | 'a.b.c'
+        'import x.y.z.*;\n@Annotation\n(\n  var = "text"\n)\nmodule //x.y.z\n/*-->*/a.b.c' | 'a.b.c'
     }
 
     @Unroll

--- a/src/test/resources/hello-annotated-module/build.gradle
+++ b/src/test/resources/hello-annotated-module/build.gradle
@@ -1,0 +1,31 @@
+plugins {
+    id 'org.beryx.jlink'
+    id 'org.javamodularity.moduleplugin' version '1.8.12'
+}
+
+repositories {
+    mavenCentral()
+}
+
+sourceCompatibility = 17
+targetCompatibility = 17
+
+dependencies {
+    implementation 'org.jspecify:jspecify:1.0.0'
+}
+
+application {
+    mainClass = 'org.beryx.modular.annotatedmodule.HelloAnnotatedModule'
+}
+jar {
+    manifest {
+        attributes 'Implementation-Title': 'helloAnnotatedModule',
+                'Main-Class': application.mainClass
+    }
+}
+
+jlink {
+    launcher {
+        name = 'helloAnnotatedModule'
+    }
+}

--- a/src/test/resources/hello-annotated-module/settings.gradle
+++ b/src/test/resources/hello-annotated-module/settings.gradle
@@ -1,0 +1,1 @@
+rootProject.name = 'helloAnnotatedModule'

--- a/src/test/resources/hello-annotated-module/src/main/java/module-info.java
+++ b/src/test/resources/hello-annotated-module/src/main/java/module-info.java
@@ -1,0 +1,7 @@
+import org.jspecify.annotations.NullMarked;
+
+@NullMarked
+module org.beryx.modular.annotatedmodule {
+    requires org.jspecify;
+    opens org.beryx.modular.annotatedmodule;
+}

--- a/src/test/resources/hello-annotated-module/src/main/java/org/beryx/modular/annotatedmodule/HelloAnnotatedModule.java
+++ b/src/test/resources/hello-annotated-module/src/main/java/org/beryx/modular/annotatedmodule/HelloAnnotatedModule.java
@@ -1,0 +1,9 @@
+package org.beryx.modular.annotatedmodule;
+
+import java.io.StringReader;
+
+public class HelloAnnotatedModule {
+	public static void main(String[] args) throws Exception {
+		System.out.println("Hello annotated module!");
+	}
+}

--- a/src/test/resources/hello-toolchain/build.gradle
+++ b/src/test/resources/hello-toolchain/build.gradle
@@ -21,8 +21,8 @@ application {
 
 java {
     toolchain {
-        languageVersion = JavaLanguageVersion.of(23)
-        vendor = JvmVendorSpec.ADOPTIUM
+        languageVersion = JavaLanguageVersion.of(24)
+        //vendor = JvmVendorSpec.ADOPTIUM
     }
 }
 

